### PR TITLE
Allow setting StorageOptions in WalletBuilder

### LINKED ISSUE #111: Extend storage functionality (encrypt/decrypt, ...)
## Description

Extend storage functionality with encryption as we currently have in the dev branch, might require some bigger changes to the storage module, because the current version is reduced to the basic functionality.

Transaction ids and output ids can't be used as keys in the db directly, otherwise any encryption is useless since the corresponding things can be found in the public

## Motivation

Privacy

## Requirements

1. Add options to set an encryption key
2. Be able to change the encryption key
(3. Function do "logout", which clears the encryption key and data in memory)

## Open questions (optional)

Might make sense to wait for https://github.com/iotaledger/wallet.rs/issues/806 before doing it or doing both at once

## Are you planning to do it yourself in a pull request?

No.

### PR DESCRIPTION
**Pull request by thibault-martinez**
Fixes https://github.com/iotaledger/iota-sdk/issues/111

### CODE REVIEW

File: sdk/src/wallet/wallet/builder.rs [hunk 1]
```diff
@@ -56,11 +56,11 @@ impl<S: SecretManage> Default for WalletBuilder<S> {
 #[derive(Debug, Clone, Serialize, Deserialize)]
 #[cfg(feature = "storage")]
 #[cfg_attr(docsrs, doc(cfg(feature = "storage")))]
-pub(crate) struct StorageOptions {
-    pub(crate) storage_path: PathBuf,
-    pub(crate) storage_file_name: Option<String>,
-    pub(crate) storage_encryption_key: Option<[u8; 32]>,
-    pub(crate) manager_store: ManagerStorage,
+pub struct StorageOptions {
```
Comment (Thoralf-M) [inline sdk/src/wallet/wallet/builder.rs]: Should probably also get serde camelCase and then requires a migration?

Comment (thibault-martinez) [reply sdk/src/wallet/wallet/builder.rs]: Yeah I was thinking about it, I also don't like the fields names

Comment (thibault-martinez) [reply sdk/src/wallet/wallet/builder.rs]: Done

File: sdk/src/wallet/wallet/builder.rs [hunk 2]
```diff
@@ -58,21 +58,22 @@ impl<S: SecretManage> Default for WalletBuilder<S> {
...
 #[cfg(feature = "storage")]
 impl Default for StorageOptions {
     fn default() -> Self {
         Self {
-            storage_path: default_storage_path().into(),
-            storage_file_name: None,
-            storage_encryption_key: None,
-            manager_store: ManagerStorage::default(),
+            path: default_storage_path().into(),
+            file_name: None,
```
Comment (Thoralf-M) [inline sdk/src/wallet/wallet/builder.rs]: This is an unused field we can remove, no?

Comment (thibault-martinez) [reply sdk/src/wallet/wallet/builder.rs]: Removed

File: sdk/CHANGELOG.md [hunk 3]
```diff
@@ -110,6 +111,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Rename `Account::prepare_mint_native_token` to `prepare_create_native_token`, `Account::prepare_increase_native_token_supply` to `prepare_mint_native_token`, `Account::prepare_decrease_native_token_supply` to `prepare_melt_native_token`;
 - Rename `MintNativeTokenParams` to `CreateNativeTokenParams`;
 - Rename `MintNativeTokenTransaction` to `CreateNativeTokenTransaction` and `PreparedMintNativeTokenTransaction` to `PreparedCreateNativeTokenTransaction` (including their corresponding DTOs);
+- Made `ManagerStorage` public and renamed it to `StorageKind`;
+- Fields of `StorageOptions` are now public;
```
Comment (Thoralf-M) [inline sdk/CHANGELOG.md]: They are not?

Comment (thibault-martinez) [reply sdk/CHANGELOG.md]: Hmm yeah, must have fucked up a merge

Comment (Thoralf-M) [reply sdk/CHANGELOG.md]: Still not public :trollface:

Comment (thibault-martinez) [reply sdk/CHANGELOG.md]: Yeah yeah I haven't done it yet, chill :trollface:

Comment (thibault-martinez) [reply sdk/CHANGELOG.md]: Done

**Review by Thoralf-M — DISMISSED** [review]

**Review by DaughterOfMars — DISMISSED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

**Pull request merged**